src/main/java/net/dries007/tfc/common/recipes/SimplePotRecipe.java

```diff
@@ -0,0 +1,239 @@
+
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Queue;
+import javax.annotation.Nullable;
+
+import com.google.gson.JsonObject;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.item.crafting.ShapedRecipe;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.Constants;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.items.ItemStackHandler;
+
+import com.mojang.serialization.JsonOps;
+import net.dries007.tfc.util.Helpers;
+
+//todo: much of the functionality is here, just needs to be finished
+public class SimplePotRecipe implements IPotRecipe
+{
+    protected final ResourceLocation id;
+    protected final HashSet<Ingredient> inputItems;
+    protected final NonNullList<ItemStack> outputItems;
+    protected final int duration;
+    protected final float temperature;
+    protected final FluidStack inputFluid;
+    protected final FluidStack outputFluid;
+
+    public SimplePotRecipe(ResourceLocation id, HashSet<Ingredient> inputItems, NonNullList<ItemStack> outputItems, FluidStack inputFluid, FluidStack outputFluid, float temperature, int duration)
+    {
+        this.id = id;
+        this.inputItems = inputItems;
+        this.outputItems = outputItems;
+        this.duration = duration;
+        this.temperature = temperature;
+        this.inputFluid = inputFluid;
+        this.outputFluid = outputFluid;
+    }
+
+    public NonNullList<ItemStack> getOutputItems()
+    {
+        return outputItems;
+    }
+
+    @Override
+    public int getDuration()
+    {
+        return duration;
+    }
+
+    @Override
+    public boolean isValidTemperature(float temperatureIn)
+    {
+        return temperatureIn >= temperature;
+    }
+
+    @Override
+    public boolean matches(FluidInventoryRecipeWrapper wrapper, World worldIn)
+    {
+        if (!wrapper.getInputFluid().isFluidEqual(inputFluid)) return false;
+
+        HashSet<Ingredient> notFound = new HashSet<>(inputItems);
+        notFound.removeIf(Ingredient::isEmpty); // clear the clutter out
+
+        HashSet<ItemStack> stacks = new HashSet<>();
+        for (int i = 0; i < wrapper.getContainerSize(); i++)
+        {
+            stacks.add(wrapper.getItem(i).copy());
+        }
+        for (ItemStack stack : stacks)
+        {
+            for (Ingredient i : notFound)
+            {
+                if (i.test(stack))
+                {
+                    notFound.remove(i);
+                    break; // removeIf not used so we can handle duplicates
+                }
+            }
+        }
+        return notFound.isEmpty();
```

Comment:
Two ingredients that match the same item stack will fail here I think. This needs to properly assign each stack to a matching ingredient. Although that quickly becomes a non trivial implementation of parsing a [unique perfect matching from a half graph](https://en.wikipedia.org/wiki/Half_graph#Matching), which... might be overboard? but consider:

spicy soup:
- 4 food ingredients
- one spicy food

that would fail under both the single-stack-matches-multiple-ingredients issue, and would potentially fail with an incorrectly labeled graph matching.

Alternatively, we restrict to one ingredient and perform a linear check on all ingredients.